src/core/qgsexpression.cpp

```diff
@@ -2187,6 +2188,38 @@ static QVariant fcnMakePolygon( const QVariantList& values, const QgsExpressionC
   return QVariant::fromValue( QgsGeometry( polygon ) );
 }
 
+static QVariant fcnMakeTriangle( const QVariantList& values, const QgsExpressionContext*, QgsExpression* parent )
+{
+  if ( values.count() < 3 )
+  {
+    return QVariant();
+  }
+
+  QgsTriangle *tr = new QgsTriangle();
+  QgsLineString* lineString = new QgsLineString();
+  lineString->clear();
+
+  Q_FOREACH ( const QVariant& value, values )
+  {
+    QgsGeometry geom = getGeometry( value, parent );
+    if ( geom.isNull() )
+      continue;
```

Comment:
What happens in the case that any of these continues are hit? Won't you be setting the exterior ring to a linestring with < 3 vertices?

I think it's better to flag an evaluation error if any of these conditions are hit.